tests/Core.nut

```diff
@@ -0,0 +1,422 @@
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+// Default params for createTest function
+defaultParams = {
+    // Options for Rocky constructor
+    "paramsRocky": {},
+    // Options for calling Rocky.Route constructor directly
+    "paramsRockyRoute": null,
+    // Options for calling Rocky.Context constructor directly
+    "paramsRockyContext": null,
+    // Use for calling Rocky.Context constructor directly. If true, then all Rocky.Context methods will be called.
+    "paramsRockyContextAdditionalUsage": false,
+    // Method to send requests
+    "method": "GET",
+    // Method to receive requests
+    // If not specified, defaultParams.method will be used
+    "methodOverride": null,
+    // If true, get post put methods will execute with Rocky.on function directly.
+    "methodStrictUsage": false,
+    // Signature to receive requests
+    "signature": "/test",
+    // Signature to send requests
+    // If not specified, defaultParams.signature will be used
+    "signatureOverride": null,
+    // Headers to send requests
+    "headers": {},
+    // Body to send requests
+    "body": "",
+    // If true, server will not respond to received requests
+    "timeout": false,
+    // If defined, set timeout param at Rocky.on|Rocky.VERB
+    "timeoutRoute": null,
+    // Expected statuscode of server's response. If real statuscode will be different, then test will be failed.
+    "statuscode": 200,
+    // Callback, that will be called, when server will receive new request. This callback (if specified) will be passed directly into Rocky.on method.
+    "cb": null,
+    // If true, defaultParams.cb will be forced to be null
+    "cbUseNull": false,
+    // Callback, that will be called, when server will receive new request. 
+    // If not specified, then server will respond with 200 statuscode.
+    // If params.cb specified, then this callback will have no affect.
+    "callback": null,
+    // Callback, that will be called, when programm will get response from the server. Usefull to verify server response.
+    // Should return true, if the test was successful, false otherwise.
+    // If not specified, then by default the test will be successful (or not, considering the statuscode).
+    "callbackVerify": null,
+    // Specifies how many times program should send requests to the server.
+    "numberOfRequests": 1,
+    // Middleware or array of middlewares for Rocky.use
+    "mwApp": [],
+    // Middleware or array of middlewares for Rocky.Route.use
+    "mw": [],

```

Comment:
Either comments don't match the variables or the variable names are misleading:

```
   // Middleware or array of middlewares for Rocky.use
   "mwApp": [],
   // Middleware or array of middlewares for Rocky.Route.use
   "mw": [],
```

if it's route mw, let's just call it `routeMiddleware`